ssn/integrated/proposed-server-specification.md

```diff
@@ -528,3 +528,119 @@ When operating a GET at http://www.w3.org/ns/ssn/systems/xyz with a Accept heade
+RewriteCond %{HTTP_ACCEPT} !(application/rdf\+xml|text/turtle)
+RewriteRule  ^(.*)$  https://www.w3.org/TR/vocab-ssn/#SOSA$1 [R=303,NE,env=html:1]
+
+RewriteCond  %{REQUEST_URI}  /ns/sosa/(ActuatableProperty|Actuation|Actuator|FeatureOfInterest|ObservableProperty|Observation|Platform|Procedure|Repeatability|Result|Sample|Sampler|Sampling|Sensor|actsOnProperty|madeByActuator|hasFeatureOfInterest|hasResult|hasResultingSample|hasSample|hosts|isActedOnBy|isFeatureOfInterestOf|isHostedBy|isObservedBy|isResultOf|isSampleOf|isSamplingResultOf|madeActuation|madeBySampler|madeBySensor|madeObservation|madeSampling|observedProperty|observes|phenomenonTime|usedProcedure|hasSimpleResult|hasResultTime)
+RewriteCond %{HTTP_ACCEPT} application/rdf\+xml
+RewriteRule  ^(.*)$  /ns/sosa/sosa.rdf [R=303,env=rdf:1]
+Header always set Content-Type "application/rdf+xml" env=rdf
+
+RewriteCond  %{REQUEST_URI}  /ns/sosa/(ActuatableProperty|Actuation|Actuator|FeatureOfInterest|ObservableProperty|Observation|Platform|Procedure|Repeatability|Result|Sample|Sampler|Sampling|Sensor|actsOnProperty|madeByActuator|hasFeatureOfInterest|hasResult|hasResultingSample|hasSample|hosts|isActedOnBy|isFeatureOfInterestOf|isHostedBy|isObservedBy|isResultOf|isSampleOf|isSamplingResultOf|madeActuation|madeBySampler|madeBySensor|madeObservation|madeSampling|observedProperty|observes|phenomenonTime|usedProcedure|hasSimpleResult|hasResultTime)
+RewriteCond %{HTTP_ACCEPT} text/turtle
+RewriteRule  ^(.*)$  /ns/sosa/sosa.ttl [R=303,env=ttl:1]
+Header always set Content-Type "text/turtle" env=ttl
+
+
+## sosa-om
+
+RewriteCond  %{REQUEST_URI}  /ns/sosa/om$
+RewriteCond %{HTTP_ACCEPT} !(application/rdf\+xml|text/html)
+RewriteRule  ^(.*)/?$  /ns/sosa/om.ttl [env=redirect:1]
+
+RewriteCond  %{REQUEST_URI}  /ns/sosa/om$
+RewriteCond %{HTTP_ACCEPT} application/rdf\+xml
+RewriteRule  ^(.*)/?$  /ns/sosa/om.rdf [env=redirect:1]
+
+RewriteCond  %{REQUEST_URI}  /ns/sosa/om$
+RewriteCond %{HTTP_ACCEPT} text/html
+RewriteRule  ^(.*)/?$  https://www.w3.org/TR/vocab-ssn/#OM_Alignment [R=303,NE,env=redirect:1]
+
+
+## sosa-oboe
+
+RewriteCond  %{REQUEST_URI}  /ns/sosa/oboe$
+RewriteCond %{HTTP_ACCEPT} !(application/rdf\+xml|text/html)
+RewriteRule  ^(.*)/?$  /ns/sosa/oboe.ttl [env=redirect:1]
+
+RewriteCond  %{REQUEST_URI}  /ns/sosa/oboe$
+RewriteCond %{HTTP_ACCEPT} application/rdf\+xml
+RewriteRule  ^(.*)/?$  /ns/sosa/oboe.rdf [env=redirect:1]
+
+RewriteCond  %{REQUEST_URI}  /ns/sosa/oboe$
+RewriteCond %{HTTP_ACCEPT} text/html
+RewriteRule  ^(.*)/?$  https://www.w3.org/TR/vocab-ssn/#OBOE_Alignment [R=303,NE,env=redirect:1]
+
```

Comment:
The rewrite rules depend on the contents of the `Accept` HTTP header, so `Accept` should be added to the `Vary` HTTP response header. Apache does not do that automatically.

In other words, I believe the following directive should be added:

```apache
Header merge Vary "Accept"
```

Reply:
Agreed, `Vary` should be added but not on the "representations" URLs:

- Vary at http://www.w3.org/ns/sosa
- Vary at http://www.w3.org/ns/sosa/Actuation
- Vary at http://www.w3.org/ns/sosa/oboe
- no Vary at http://www.w3.org/ns/sosa/sosa.ttl
- no Vary at http://www.w3.org/ns/sosa/oboe.rdf

So I think this should do the job:

```
Header always set Vary "Accept" env=ttl
Header always set Vary "Accept" env=rdf
Header always set Vary "Accept" env=html
```

Reply:
Right. Note I will use `merge` and not `set` in my case, not to destroy other variation rules applied upper in the hierarchy. In particular, our server returns a `Vary: upgrade-insecure-requests` HTTP header that needs to stay.